packages/typespec-client-generator-core/test/methods/lro.test.ts

```diff
@@ -816,6 +816,97 @@ describe("data plane LRO templates", () => {
     );
     strictEqual(lroMetadata.pollingStep.responseBody, analyzeOperationModel);
   });
+
+  // https://github.com/Azure/typespec-azure/issues/2325
+  it("LroMetadata final result anonymous model", async () => {
+    await runner.compileWithVersionedService(`
+      alias ServiceTraits = NoRepeatableRequests &
+        NoConditionalRequests &
+        NoClientRequestId;
+
+      alias apiOperations = Azure.Core.ResourceOperations<ServiceTraits>;
+
+      @doc("Create an instruction.")
+      @pollingOperation(
+        OperationProgress.getOperationResult
+      )
+      op update is apiOperations.LongRunningResourceCreateOrUpdate<Instruction>;
+
+      @resource("instruction")
+      model Instruction {
+        @key
+        @visibility(Lifecycle.Read)
+        id: string;
+
+        @visibility(Lifecycle.Update)
+        instructionId: string;
+      }
+
+      @doc("Operation Status")
+      @lroStatus
+      union OperationStatusValue {
```

Comment:
Is it needed to be defined for this test? Can we use the `Foundations.OperationState` in Core?
Or `OperationResultQuery` itself can be replaced by some model in Core?

Feel that both of them is not relevant to the test case. Only the definition of `Instruction` model matters.